src/VisualStudio/Core/Next/Remote/ServiceHubRemoteHostClient.cs

```diff
@@ -130,6 +131,16 @@ private void OnRpcDisconnected(object sender, JsonRpcDisconnectedEventArgs e)
                     var descriptor = new ServiceDescriptor(serviceName) { HostGroup = hostGroup };
                     return await client.RequestServiceAsync(descriptor, cancellationToken).ConfigureAwait(false);
                 }
+                catch (OperationCanceledException)
+                {
+                    // if it is our own cancellation token, then rethrow
+                    // otherwise, let us retry.
```

Comment:
This "let us retry" is *exactly* why StreamJsonRpc and HubClient don't. As we have multiple levels in our callstack here, and generally the scenario owner will have a retry, if the lower-level libraries had retries as well, they would multiply their effect, in the end making the user wait *much* longer than the scenario owner intended.